src/app/(pages)/fhirServers/page.tsx

```diff
@@ -306,17 +307,20 @@ const FhirServers: React.FC = () => {
     disableCertValidation: boolean,
     authData: AuthData,
   ) => {
-    const supportsMatch = await checkFhirServerSupportsMatch(
+    const { supportsMatch, fhirVersion } = await checkFhirServerSupportsMatch(
       hostname,
       disableCertValidation,
       authData,
     );
+
+    setFhirVersion(fhirVersion);
     setPatientMatchData((prev) => ({
       enabled: prev?.enabled ?? false,
-      onlySingleMatch: prev?.onlySingleMatch ?? false,
-      onlyCertainMatches: prev?.onlyCertainMatches ?? false,
-      matchCount: prev?.matchCount ?? 1,
-      supportsMatch,
+      onlySingleMatch: false,
+      onlyCertainMatches:
+        fhirVersion && fhirVersion.startsWith("6") ? false : true,
```

Comment:
Should false and true be swapped since R6 supports it and other versions do not?